src/components/src/main/java/org/apache/jmeter/assertions/JSONPathAssertion.java

```diff
@@ -164,8 +166,12 @@ private boolean arrayMatched(JSONArray value) {
     private boolean isEquals(Object subj) {
         if (isUseRegex()) {
             String str = objectToString(subj);
-            Pattern pattern = JMeterUtils.getPatternCache().getPattern(getExpectedValue());
-            return JMeterUtils.getMatcher().matches(str, pattern);
+            if (useJavaRegex) {
+                return java.util.regex.Pattern.matches(getExpectedValue(), str);
+            } else {
+                Pattern pattern = JMeterUtils.getPatternCache().getPattern(getExpectedValue());
```

Comment:
I think we should also have a Cache for java Patterns as compilation has a performance penalty no ? 
Could we have a kind of factory which would give the implementation based on useJavaRegex ?

Reply:
I can add a factory to JMeterUtils like the old one. We include caffeine, so a cache should be added real easy. For starting I chose not to add it, as it is more work and I wanted to have a working state first.

Reply:
Added a cache with [1d30005](https://github.com/apache/jmeter/pull/700/commits/1d3000597810a7e5c379cabad00c1369002a0c1f) Now, we should use it and measure, if it is performing well :)

Reply:
I have done some simple performance tests with the ResponseAssertion.
95% response time/99% response time/Transactions per second for *X* number of threads.

|Threads|Oro|Java (no cache)|Java (with cache)|
|-----------|-----|---------------------|-----------------------|
|1     |1.0/1.0/4266.62|1.0/1.0/5763.90|1.0/1.0/5722.43|
|10   |1.0/5.0/12939.44|1.0/5.0/12731.02|1.0/5.0/12939.44|
|100 |6.0/58.99/9862.14|6.0/77.0/11897.78|5.0/79.0/11887.85|
|1000|9.0/259.0/7867.21|3.0/56.0/9773.83|4.0/36.0/10056.35|

The test was one JSR-223 Groovy Sampler generating a long text and one ResponseAssertion asserting a contains (of the last line).

Reply:
Nice !